### PR TITLE
[Feat]: Displaying Intern Pairings in Week card

### LINKED ISSUE #29: [Feat]: Create a pairing util
## Description

Create a util function that can make pairs out of an array. A good resources is an array of [tuples](https://flow.org/en/docs/types/tuples/)! (Note: this resource references typescript, but you can do the same thing without the typing.

## Tasks

- [x] Create a util function to pair up the interns
- [x] Consider when we need to make a pair of 3 and a pair of 2 and implement the edge case

### Notes
Pairing should be the last operation we should do before we display the intern information, so make sure the randomizing + filter happens before pairing things (this doesn't mean we can't do this function without just that this is the last stop for the algorithm before displaying that information)

### PR DESCRIPTION
**Pull request by jacky-p**
**When you open your pull requests, remove the lines that start and end with underscores (\_)**

## Changes
_List Changes Introduced by this PR_
1. Added display for dummy data that is added to the table of the Week card, dummy data and ice breaker questions will be displayed when the 'Generate Schedule' button is clicked 

**Display before button is clicked**
Screen Shot 2024-10-01 at 2 36 21 PM

**Display after button is clicked**
Screen Shot 2024-10-01 at 2 37 18 PM

2. Also added randomized pairing functionality, so that every time the generate button is clicked new randomized parings and new questions are displayed.
**Display after clicking the generate button again**
Screen Shot 2024-10-01 at 2 37 05 PM

## Purpose
Will display the pairings generated in the Week card when the generate button is clicked. It will also display varying ice breaker questions when clicked as well.

## Approach
Given a JSON file with the list of interns this feature works by fetching that intern data from the JSON and converting it into an array, shuffling the array (randomizing the array), and paring them off. Afterwards the data (the paired of array, which is an array of tuples) is passed into the display function where then the info is displayed within a table. 

The function iterates through each pair where we create a row element and populate it with the interns' info. Following how the table was structured in the HTML, created the elements and nested them in order to display the information correctly. All elements are then appended accordingly.

## Pre-Testing TODOs
_What needs to be done before testing_
- [x] Merge buttons
- [x] Incorporate its function to the buttons
- [x] Add randomized pairing function
- [x] Odd number of interns compatibility

## Testing Steps
1. Click on "Generate Schedule" and the randomized intern pairing should show up as well as new questions.

## Learning
Alex's zoom meeting about tables was very helpful.
[How to Make a Table from a JSON Object](https://us02web.zoom.us/rec/share/AtDumOd1-XLehJWpfZnBKV0MxXt8_uoVFynfxtZz-pcQaTlMo8BrIVQwZ1GgqFtU.uopehY0QGyf1Y7Vh)
[Arrays in Javascript](https://www.w3schools.com/js/js_arrays.asp)

Closes #29

### CODE REVIEW

File: src/scripts/organizeInternInfo.js
Comment (elimanzo) [inline src/scripts/organizeInternInfo.js:1]: I love this good job!

File: src/util/pair.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export default function pair(arr) {
+  const pairsArray = []; //return array with the pairings
+  const arrLast = arr.length - 1;
+
```
Comment (elimanzo) [inline src/util/pair.js:4]: lets have some boundaries checking if the array is empty, it just returns an empty arr;

File: src/scripts/organizeInternInfo.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+export default function organizeInternInfo(intern) {
+  const col = document.createElement("td"); //Create column for intern
+  const internInfo = document.createElement("div"); //Column info div
+  internInfo.className = "intern-pill-name-location";
+
+  const pill = document.createElement("div"); //Department pill div
+  pill.className = "pill";
+  pill.innerHTML = "<b>" + intern.department + "</b>";
```
Comment (elimanzo) [inline src/scripts/organizeInternInfo.js:8]: this is a nit but you can use [template strings ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)for this 
```suggestion
  pill.innerHTML = `<b>${intern.department}</b>`;
```

**Review by elimanzo — CHANGES_REQUESTED** [review]
Good job! I have couple of nits and I think this is ready to go!

File: src/app.js [hunk 3]
```diff
@@ -26,5 +29,46 @@ document.addEventListener("DOMContentLoaded", () => {
...
+    displayInternPairs();
+    displayQuestions();
+  });
+
+  function displayInternPairs() {
+    let count = 0;
+    const table = document.getElementById("interns-table");
+    table.innerHTML = ""; //clear out any previous pairings
+
+    internPairs.forEach((pair) => {
```
[the remark this answers is not in the document]
Comment (jacky-p) [reply src/app.js:55]: Oh I see, just replaced `count` with `index` from the forEach.

File: src/app.js [hunk 4]
```diff
@@ -26,5 +29,46 @@ document.addEventListener("DOMContentLoaded", () => {
...
+      const groupNum = document.createElement("td"); //Group num column
+      groupNum.textContent = "Group " + count;
+      row.appendChild(groupNum);
+
+      //add intern info columns
+      for (let i = 0; i < pair.length; i++) {
+        console.log(pair[i]);
+        const intern = organizeInternInfo(pair[i]);
+        row.appendChild(intern);
+      }
```
[the remark this answers is not in the document]
Comment (jacky-p) [reply src/app.js:68]: Ohh I wasn't aware of that function, that's helpful, thanks!

File: src/app.js [hunk 5]
```diff
@@ -26,5 +29,46 @@ document.addEventListener("DOMContentLoaded", () => {
     }
   }
 
-  displayQuestions();
+  //displayQuestions();
+
+  const generateButton = document.getElementById("schedule-button");
+
+  let internPairs = [];
+  async function pairInterns() {
```
[the remark this answers is not in the document]
Comment (jacky-p) [reply src/app.js:37]: Ok!

File: src/scripts/organizeInternInfo.js
Comment (jacky-p) [reply src/scripts/organizeInternInfo.js:1]: Thank you!

File: src/app.js [hunk 6]
```diff
@@ -26,5 +29,41 @@ document.addEventListener("DOMContentLoaded", () => {
     }
   }
 
-  displayQuestions();
+  //displayQuestions();
```
Comment (elimanzo) [inline src/app.js:32]:
```suggestion
```
we can remove these comments

File: src/app.js [hunk 7]
```diff
@@ -26,5 +29,41 @@ document.addEventListener("DOMContentLoaded", () => {
...
+
+  const generateButton = document.getElementById("schedule-button");
+
+  //let internPairs = [];
```
Comment (elimanzo) [inline src/app.js:36]: we can remove this too :)
```suggestion
```

**Review by elimanzo — APPROVED** [review]
Just remove the comments and we're good to go!

**Pull request merged**